lib/livebook/application.ex

```diff
@@ -223,35 +226,47 @@ defmodule Livebook.Application do
     end
   end
 
-  defp create_teams_hub() do
+  defp parse_teams_hub() do
     teams_key = System.get_env("LIVEBOOK_TEAMS_KEY")
     auth = System.get_env("LIVEBOOK_TEAMS_AUTH")
 
     cond do
       teams_key && auth ->
-        Application.put_env(:livebook, :teams_auth?, true)
-
-        hub =
+        fun =
           case String.split(auth, ":") do
             ["offline", name, public_key] ->
-              create_offline_hub(teams_key, name, public_key)
+              Application.put_env(:livebook, :teams_auth, :offline)
+              fn -> create_offline_hub(teams_key, name, public_key) end
 
             ["online", name, org_id, org_key_id, agent_key] ->
-              create_online_hub(teams_key, name, org_id, org_key_id, agent_key)
+              Application.put_env(:livebook, :teams_auth, :online)
+              fn -> create_online_hub(teams_key, name, org_id, org_key_id, agent_key) end
 
             _ ->
               Livebook.Config.abort!("Invalid LIVEBOOK_TEAMS_AUTH configuration.")
           end
 
-        Application.put_env(:livebook, :apps_path_hub_id, hub.id)
+        case Application.fetch_env(:livebook, :identity_provider) do
+          {:ok, _} ->
+            :ok
+
+          :error ->
+            Application.put_env(
+              :livebook,
+              :identity_provider,
+              {:zta, Livebook.ZTA.LivebookTeams, :unused}
+            )
+        end
```

Comment:
We only want to set it for online, because offline inherently has no connection, right?

Reply:
Also, because we don't have all the information to authenticate with Teams

Reply:
We are still setting it for offline, no?

Reply:
Yeah, it's better to remove for now, until we have a solution for this use case.